CorsixTH/Lua/app.lua

```diff
@@ -1680,60 +1680,136 @@ function App:loadLuaFolder(dir, no_results, append_to)
+  savegame_version = savegame_version or self:getCurrentVersion()
+  local release_data
+  for i = #release_table, 1, -1 do
+    local release = release_table[i]
+    if release.version == savegame_version then
+      release_data = release
+      break
+    elseif (release.version - savegame_version) < 0 then
+      -- we're not on a release version
+      local develop = shallow_clone(release) -- prevent recursion
+      develop.version = savegame_version
+      develop.patch = release.patch .. "-dev" .. savegame_version
+      release_data = develop
+      break
+    end
+  end
+  return release_data
+end
+
+--! Provides the release (or development) string for a given savegame version
+--!param savegame_version (number) The version to look up
+--!return A string in the format "v<major>.<minor>.<revision>[patch]"
+--! For development builds, the savegame version is appended as a patch e.g. 'dev213'
+function App:getReleaseString(savegame_version)
+  local release = self:getReleaseData(savegame_version)
+  local release_string = "v" .. release.major .. "." .. release.minor .. "." ..
+      release.revision
+  if string.len(release.patch) > 0 then
+    release_string = release_string .. release.patch
+  end
+  return release_string
+end
+
+--! Reports a difference between two versions based on requested methodology.
+--!param version_a (number or table) The first (usually newer) version to test
+--!param version_b (number or table) The second (usually older) version to test
+--!param method (string) What method to compare by
+--- method(release) reports the difference between the matching releases in steps,
+--- revisions are not counted.
+--- For development builds it will use the base release it started from when using
+--- the release method
+--- method(version) reports the difference between two savegame versions
+--!return The step difference between release a and release b for release method
+--- or The raw savegame version difference for version method
+function App:compareVersions(version_a, version_b, method)
+  assert((type(version_a) == "table" or type(version_a) == "number") and
+      (type(version_b) == "table" or type(version_b) == "number"),
+      "Compare requires savegame version or the version table to function")
+  assert(method == "release" or method == "version",
+      "Not using a valid compare method")
+
+  if method == "release" then
+    local function countBackward(version_to_check)
+      local step = 0
+      for i = #release_table, 1, -1 do
+        local release = release_table[i]
+        if release.version == version_to_check then
+          break
+        elseif (release.version - version_to_check) < 0 then
+          -- we're not on a release version
+          if step == 0 then step = 1 break end -- working from current development
+          break
+        end
+        step = (release.revision == 0 and string.len(release.patch) == 0)
+            and step - 1 or step
```

Comment:
```suggestion
        if release.revision == 0 and release.patch == "" then
          step = step - 1
        end
```

Reply:
The equality check is cheaper and more straight forward, I think the if condition is clearer here.